browser/ui/views/omnibox/brave_omnibox_result_view.cc

```diff
@@ -42,6 +45,10 @@ void BraveOmniboxResultView::ResetChildren() {
     brave_search_promotion_view_ = nullptr;
   }
 
+  if (leo_match_label_) {
+    RemoveChildViewT(leo_match_label_.ExtractAsDangling());
```

Comment:
@cdesouza-chromium is this the correct solution?

When I do this:

```cpp
RemoveChildViewT(leo_match_label_);
leo_match_label_ = nullptr;
```

I get a dangling ptr exception